core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -343,21 +345,78 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
     /**
      * Deletes the internal topic partition info if delete flag is set as true.
      *
-     * @param topicPartition topic partition to be stopped.
+     * @param topicPartitions topic partitions that needs to be stopped.
      * @param delete         flag to indicate whether the given topic partitions to be deleted or not.
      */
-    public void stopPartitions(TopicPartition topicPartition, boolean delete) {
+    public void stopPartitions(Set<TopicPartition> topicPartitions,
```

Comment:
please consider invalidating the segments belonging to this partition from RemoteIndexCache when we are stopping partitions.

Reply:
RemoteIndexCache does not maintains the indexes by partition instead by segment-uuid. Traversing the cache to remove those indexes will be expensive. Since, the `RemoteIndexCache` is a LRU cache, those indexes will be expired eventually.